clinguin_ui/ui_menu_bar.lp

```diff
@@ -0,0 +1,183 @@
+
+        %TODO on click show modal with entry option to select a file path
+
+
+        elem(save_modal,modal,w).
+        attr(save_modal,title,"Save file").
+        % attr(save_modal,size,"sm").
+            elem(c_save, container, save_modal).
+            attr(c_save, flex_direction, "column").
+            attr(c_save, class, "align-items-end").
+
+                elem(b1, button, c_save).
+                when(b1, click, call, download(show_prg,_context_value(file_name,str,"ooasp_configuration.lp"))).
+                attr(b1, label, "Save").
+                attr(b1, class, "m-1").
+                attr(b1, class, "btn-primary").
+                attr(b1, icon, "fa-download").
+
+                elem(t1, textfield, c_save).
+                attr(t1, placeholder, "ooasp_configuration.lp").
+                attr(t1, width, 450).
+                when(t1, input, context, (file_name, _value)).
+
+    elem(end_btn,button,menu_bar):-_clinguin_browsing.
+    attr(end_btn,label,"End browsing"):-_clinguin_browsing.
+    attr(end_btn,icon,"fa-xmark"):-_clinguin_browsing.
+    attr(end_btn,class,("btn-success")):-_clinguin_browsing.
+    when(end_btn,click,call,stop_browsing):-_clinguin_browsing.
+
+    elem(find_btn,button,menu_bar).
+    attr(find_btn,label,"Find Incrementally").
+    attr(find_btn,icon,"fa-gears").
+    attr(find_btn,class,("btn-outline-secondary")).
+    when(find_btn,click,call,find_incrementally).
+
+    elem(suggest_btn,button,menu_bar):- _clinguin_external(check_potential_cv,true).
+    attr(suggest_btn,label,"Next Solution") :- _clinguin_external(check_potential_cv,true).
+    attr(suggest_btn,icon,"fa-forward") :- _clinguin_external(check_potential_cv,true).
+    attr(suggest_btn,class,("btn-outline-success")):- _clinguin_external(check_potential_cv,true).
+    % attr(suggest_btn,class,("btn-success")) :- _clinguin_external(check_potential_cv,true).
+    when(suggest_btn,click,call, next_solution):- _clinguin_external(check_potential_cv,true).
+
+    elem(select_btn,button,menu_bar) :- _clinguin_browsing.
+    attr(select_btn,label,"Select solution") :- _clinguin_browsing, _clinguin_external(check_potential_cv,true).
+    attr(select_btn,label,"Select suggestion") :- _clinguin_browsing, _clinguin_external(check_potential_cv,false).
+    attr(select_btn,class,("btn-outline-success")) :- _clinguin_browsing.
+    attr(select_btn, icon, "fa-hand-pointer") :- _clinguin_browsing.
+    when(select_btn, click, call, select(show_prg)) :- _clinguin_browsing.
+
+elem(unsat_msg, message, w):-_clinguin_unsat.
+attr(unsat_msg, message, "No solution with the current objects"):-_clinguin_unsat.
+attr(unsat_msg, type, "error"):-_clinguin_unsat.
+attr(unsat_msg, title, "Unsatisfiable"):-_clinguin_unsat.
+
+    % elem(check_btn,button,menu_bar).
+    % attr(check_btn,label,"Check").
+    % attr(check_btn,icon,"fa-magnifying-glass").
+    % attr(check_btn,class,("btn-outline-danger")).
```

Comment:
Why is this commented-out?

Reply:
We decided not to use it and do the checks automatically, it is just a leftover.
Now removed.